shaders/src/light_punctual.fs

```diff
@@ -211,18 +211,15 @@ void evaluatePunctualLights(const MaterialInputs material,
         if (light.NoL > 0.0) {
             if (light.castsShadows) {
                 uint shadowIndex = light.shadowIndex;
-                highp vec4 shadowPosition;
                 if (light.type == LIGHT_TYPE_POINT) {
                     // point-light shadows are sampled from a direction
                     highp vec3 r = getWorldPosition() - light.worldPosition;
-                    highp uint face = 0u;
-                    // getShadowPosition returns zLight which is needed for PCSS/DPCF
-                    shadowPosition = getShadowPosition(r, shadowIndex, light.zLight, face);
+                    highp uint face = getPointLightFace(r);
                     shadowIndex += face;
-                } else {
-                    // getShadowPosition needs zLight for applying the normal bias
-                    shadowPosition = getShadowPosition(false, shadowIndex, 0u, light.zLight);
+                    light.zLight = dot(shadowUniforms.shadows[shadowIndex].lightFromWorldZ,
+                            vec4(getWorldPosition(), 1.0));
                 }
+                highp vec4 shadowPosition = getShadowPosition(false, shadowIndex, 0u, light.zLight);
```

Comment:
`getShadowPosition` calls `getSpotLightSpacePosition`, should we rename the later to `getPointLightSpacePosition`?

Reply:
well, in the shader there is no concept of "point light position". Once we have the proper "shadowIndex", it's just like a spotlight -- so it calls getSpotLightSpacePosition(). We could rename it to `getDynamicLightPosition()` (dynamic as opposed to directional).